fvm/evm/types/result.go

```diff
@@ -49,6 +56,9 @@ func (res *Result) VMErrorString() string {
 // and for each log, Address, Topics, Data (consensus fields)
 // During execution we also do fill in BlockNumber, TxIndex, Index (event index)
 func (res *Result) Receipt() *gethTypes.Receipt {
+	if res.Invalid() {
```

Comment:
I wonder if not having a receipt for invalid transactions is correct since they do consume gas?

Reply:
Thats an EVM-expected behaviour that we can't change easily. For example, in the case of a nonce mismatch (ahead of state), there should never be a state change or receipt otherwise the 3rd party apps will break after resubmission of the transaction when the nonce becomes the right number.